CMakeLists.txt

```diff
@@ -5,6 +5,53 @@ project(
     LANGUAGES CXX
 )
 
+# dependencies
+include(FetchContent)
+if (POLICY CMP0135)
+    cmake_policy(SET CMP0135 OLD)
+endif ()
+set(fetch_content_deps googletest)
+
+set(fetch_content_supported_flags "")
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.24)
+    list(APPEND fetch_content_supported_flags OVERRIDE_FIND_PACKAGE)
+endif ()
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.25)
+    list(APPEND fetch_content_supported_flags SYSTEM)
+endif ()
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
+    list(APPEND fetch_content_supported_flags EXCLUDE_FROM_ALL)
+endif ()
+
+# Using CMake >= 3.24 we override find_package and it "just works". For older versions we simulate the same behavior by
```

Comment:
Can we automatically test this for different CMake versions in our GitHub action, i.e., add different CMake versions to our build matrix?